src/site/content/en/react/how-to-use-amp-in-nextjs-and-why/index.md

```diff
@@ -0,0 +1,272 @@
+---
+layout: post
+title: How to use AMP in Next.js (and why)
+authors:
+  - houssein
+subhead: |
+  Add AMP to any of your Next.js pages to guarantee fast page loads
+date: 2019-10-17
+draft: true
+---
+
+[AMP](https://amp.dev) (Accelerated Mobile Pages) is a web component framework that guarantees
+that your pages will load fast. [Next.js][intro] has built-in support for AMP.
+
+## What will you learn?
+
+This guide first briefly describes [how AMP guarantees fast page loads](#overview), then lets you
+try out [the two ways that you can use AMP in a Next.js app](#strategies), then helps you [decide
+which approach is best for you](#guidance).
+
+The intended audience for this guide is a web developer who has decided to use Next.js but is
+unsure of whether to support AMP. 
+
+{% Aside %}
+  This guide was not written for web developers who have decided to use AMP but are unsure of what
+  framework to use. We will note briefly however that Next.js could be a good choice because it
+  supports [AMP server-side rendering](https://amp.dev/documentation/guides-and-tutorials/optimize-and-measure/server-side-rendering/)
+  and makes it easy to serve AMP content without introducing a lot of complexity into your codebase.
+{% endAside %}
+
+## How AMP guarantees fast page loads {: #overview }
+
+AMP has two main strategies for guaranteeing fastness:
+
+* **AMP HTML**: A restricted form of HTML that makes certain optimizations mandatory and prohibits
+  architectural patterns that lead to slowness. See [How AMP works][how] for a high-level
+  overview of the optimizations and restrictions.
+* **AMP Cache**: A content cache used by some search engines, such as Google and Bing, that uses
+  [prerendering] to speed up page loads. See [Why AMP Caches exist][why] to learn more about the
+  benefits and tradeoffs of the caches and [How does my AMP page get cached?][how] to understand
+  how your AMP pages get into the caches.
+
+## How you can use AMP in Next.js {: #strategies }
+
+There are two ways to use AMP in Next.js:
+
+* The [**Hybrid AMP** approach](#hybrid) lets you create an accompanying AMP version of any
+  Next.js page.
+* The [**AMP-only** approach](#amponly) lets you make AMP the only option for a page.
+
+{% Aside 'caution' %}
+  Whenever you serve AMP pages, you won't be able to run React components client-side because
+  React components are not valid AMP components.
+{% endAside %}
```

Comment:
@alankent in the email thread you said something along these lines. Does this phrasing capture your point?

Reply:
Might help to make things more explicit about what Next.js actually does? Something like:

`Whenever you serve AMP pages with Next.js, React components do not get rendered on the client. React components are not valid AMP components and Next.js converts everything to valid AMP."

Wording could probably be better, and I'm sure @alankent will have a better suggestion.

Reply:
"converts everything to valid AMP" is not worded correctly - Next.js does not convert anything - the developer has to do the work of generating AMP output. With AMP you are using Next.js as a templating engine on the server to generate AMP pages (and API hosting etc). You are creating a React client app at all. I worry slightly that people might get the wrong idea, but worth broader discussion with others before committing to a path. Need to think about this a little more - is there a use case of tacking AMP on the side of a React application? (My normal mental model is "I want to create an AMP page, Next.js is a useful templating engine to do it.) I think worth discussing / thinking a bit more first to make sure the message is appropriate/useful.